tests/csapi/registration_test.go

```diff
@@ -0,0 +1,45 @@
+package csapi_tests
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+	"github.com/tidwall/gjson"
+)
+
+func Test11Register(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: registration accepts non-ascii passwords
+	t.Run("Registration accepts non-ascii passwords", func(t *testing.T) {
+		reqJson := json.RawMessage(`{"username": "test_user", "password": "übers3kr1t", "device_id": "xyzzy", "initial_device_display_name": "display_name"}`)
+		resp := unauthedClient.DoFunc(t, "POST", []string{"_matrix", "client", "r0", "register"}, client.WithRawBody(reqJson))
```

Comment:
Prefer using `map[string]interface{}` over raw JSON bodies as that will handle escaping for you. You do this further down with `reqBody` and `client.WithJSONBody`